### PR TITLE
created vibrance function

### PR DESCRIPTION
**Pull request by mkeyim**

### CODE REVIEW

**Comment by mkeyim** [comment]
my test functions are not working, any insight in how to making them work would be great! I cant figure out how to call them

File: picfixPy/vibrance.py [hunk 1]
```diff
@@ -1,20 +1,54 @@
-def vibrance(input_img, intensity=5, output_img):
-    
+def vibrance(input_img, intensity=0, display = False, output = ''):
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py]: Good idea for thinking of the `display` argument, I will add this argument to my function as well.

Comment (GeorgeJJW) [inline picfixPy/vibrance.py:1]: Good idea for thinking of the `display` argument, I will add this argument to my function as well.
One thing I'll like to point out here is try not to change the names of the other arguments at this stage. Unless you think our old argument names are horrid, changing the argument names of your function at this stage means you have to change the argument names throughout the entire test suite for this function! Just a thought. I recommend just keep the old argument names for consistency of other functions sake.

I think we also need to add the dependencies of the function on top of the function as well? Unless you don't even have dependencies then very nice!

```suggestion
# dependencies
import skimage.io

def vibrance(input_img, intensity=5, display=False, output_img=''):
```

File: picfixPy/vibrance.py [hunk 2]
```diff
@@ -1,20 +1,54 @@
...
-    an image file in .png format with specified vibrance in specified output path
+    an image file in .jpg, .jpeg, .png, .tiff format with specified vibrance in specified output path
 
-    Example
-    ============
-    vibrance("../img/image.png", 2, "../img/vibrance_image.png")
     '''
+    output_image = np.ndarray(input_img.shape, dtype = 'uint8')
+
+    for i in range(len(input_img)):
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:21]: Instead of using a nested for loop, maybe for next week we can look into vectorizing our code (ie taking advantage of the elementwise operation of numpy), so that the function still runs fast on high res images, just a suggestion.

File: picfixPy/vibrance.py [hunk 3]
```diff
@@ -1,20 +1,54 @@
...
+            if s < 0.0:
+                s = 0.0
+            else:
+                s = s
+
+            rgb = colorsys.hsv_to_rgb(h, s, v)
+
+            output_image[i][j][0] = rgb[0]
+            output_image[i][j][1] = rgb[1]
+            output_image[i][j][2] = rgb[2]
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:48]: Great job figuring out and implementing the vibrance feature!

File: picfixPy/vibrance.py [hunk 4]
```diff
@@ -1,20 +1,54 @@
...
 
     Input
     ============
-    input_img -- string of path for an image file in .png format
-    intensity -- integer in range [0,10] to specificy intensity of vibrance enhancement applied to the image,
-                 defaults to 5.
-    output_img -- string path for output .png image with adjusted vibrance
+
+    img -- string. Path for an image file in .jpg, .jpeg, .png, .tiff format.
+    intensity -- integer/float in range [-10,10]. Specificies intensity of vibrance applied to image.
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:10]: Interesting choice for supporting negative intensity values again, I don't think I'll do this for contrast though, because adding negative intensity mean writing more test cases :) I'm fine with having negative intensity values for the vibrance function as long as you can set up the appropriate test cases for it.
```suggestion
    intensity -- integer/float in range [-10,10]. Specifies intensity of vibrance applied to image. Defaults to 5.
```

File: picfixPy/vibrance.py [hunk 5]
```diff
@@ -1,20 +1,54 @@
...
+    Increase or decrease the vibrance of an image, given the intensity specificication.
 
     Input
     ============
-    input_img -- string of path for an image file in .png format
-    intensity -- integer in range [0,10] to specificy intensity of vibrance enhancement applied to the image,
-                 defaults to 5.
-    output_img -- string path for output .png image with adjusted vibrance
+
+    img -- string. Path for an image file in .jpg, .jpeg, .png, .tiff format.
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:9]: Catch a minor inconsistency here with the function documentation. I am still quite uncomfortable supporting non-png image format as input at this stage, before we code up the R functions. I suggest keeping things simple and stick with .png images for now. Just my two cents.
```suggestion
    input_img -- string. Path for an image file.
```

File: picfixPy/vibrance.py [hunk 6]
```diff
@@ -1,20 +1,54 @@
...
+    intensity -- integer/float in range [-10,10]. Specificies intensity of vibrance applied to image.
+    display -- bool. If True, output image will display
+    output -- string path of output image
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:12]: A related suggestion of change due to an earlier suggestion.
```suggestion
    output_img -- string path of output image.
```

File: picfixPy/vibrance.py [hunk 7]
```diff
@@ -1,20 +1,54 @@
...
     Input
     ============
-    input_img -- string of path for an image file in .png format
-    intensity -- integer in range [0,10] to specificy intensity of vibrance enhancement applied to the image,
-                 defaults to 5.
-    output_img -- string path for output .png image with adjusted vibrance
+
+    img -- string. Path for an image file in .jpg, .jpeg, .png, .tiff format.
+    intensity -- integer/float in range [-10,10]. Specificies intensity of vibrance applied to image.
+    display -- bool. If True, output image will display
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:11]: Adding a period, just being anal :)
```suggestion
    display -- bool. If True, output image will display. Defaults to False.
```

File: picfixPy/vibrance.py [hunk 8]
```diff
@@ -1,20 +1,54 @@
...
+    output -- string path of output image
 
     Returns
     ============
-    an image file in .png format with specified vibrance in specified output path
+    an image file in .jpg, .jpeg, .png, .tiff format with specified vibrance in specified output path
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:16]: Here the image types become less important, because users can just specify the type of the image they want using the `output` argument, I agree, we can relax the image type restrictions here.

File: picfixPy/vibrance.py [hunk 9]
```diff
@@ -1,20 +1,54 @@
...
 
-    Example
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:17]: I suggest keeping the example for the documentation.
```suggestion
    Example
    ============
    vibrance("../img/input.png", 5, display = True, "../img/output.png")
```

File: picfixPy/vibrance.py [hunk 10]
```diff
@@ -1,20 +1,54 @@
...
+                s = s
+
+            rgb = colorsys.hsv_to_rgb(h, s, v)
+
+            output_image[i][j][0] = rgb[0]
+            output_image[i][j][1] = rgb[1]
+            output_image[i][j][2] = rgb[2]
+
+    if display == True:
+        imshow(output_image)
```
Comment (GeorgeJJW) [inline picfixPy/vibrance.py:51]: Nice way of implementing the display argument!

File: picfixPy/vibrance.py (hunk 1 above)
Comment (GeorgeJJW) [reply picfixPy/vibrance.py]: I think we also need to add the dependencies of the function on top of the function as well? Unless you don't even have dependencies then very nice!

**Review by GeorgeJJW — CHANGES_REQUESTED** [review]
- Great job implementing the vibrance feature!

File: picfixPy/vibrance.py (hunk 1 above)
Comment (mkeyim) [reply picfixPy/vibrance.py:1]: I dont think I understand what you mean by change the names of other arguments, what arguments are changed?

File: picfixPy/vibrance.py (hunk 2 above)
Comment (mkeyim) [reply picfixPy/vibrance.py:21]: For sure! If you manage to implement it in yours, definitely let me know.

File: picfixPy/vibrance.py (hunk 4 above)
Comment (mkeyim) [reply picfixPy/vibrance.py:10]: I got the greyscale test ready to go!

File: picfixPy/vibrance.py (hunk 1 above)
Comment (GeorgeJJW) [reply picfixPy/vibrance.py:1]: oh I am just referring to the argument `output_img` being changed to `output ` here.

Comment (mkeyim) [reply picfixPy/vibrance.py:1]: ah, I'm ok with either, easy fix!
